ios/FluentUI/Bottom Commanding/BottomCommandingController.swift

```diff
@@ -534,12 +529,41 @@ open class BottomCommandingController: UIViewController {
         }
     }
 
-    private func updateSheetExpandedContentHeight() {
-        if let bottomSheetController = bottomSheetController {
-            bottomSheetController.preferredExpandedContentHeight = estimatedTableViewHeight + Constants.BottomSheet.expandedContentTopMargin
+    /// Recalculates header top margin constraint and updates the `collapsedContentHeight` and `isExpandable` properties of the sheet controller.
+    private func updateSheetHeaderSizingParameters() {
+        guard let bottomSheetController = bottomSheetController else {
+            return
+        }
+        bottomSheetController.isExpandable = isExpandable
+
+        let maxHeroItemHeight = heroCommandStack.arrangedSubviews.map { $0.intrinsicContentSize.height }.max() ?? Constants.defaultHeroButtonHeight
+        let headerHeightWithoutBottomWhitespace = Constants.BottomSheet.headerTopMargin + maxHeroItemHeight
+
+        // How much more whitespace is required at the bottom of the sheet header
+        let requiredBottomWhitespace = max(0, Constants.BottomSheet.headerHeight - headerHeightWithoutBottomWhitespace)
+
+        // The safe area inset can fulfill some or all of our bottom whitespace requirement.
+        // This is how much more we need, taking the inset into account.
+        let reducedBottomWhitespace = max(0, requiredBottomWhitespace - view.safeAreaInsets.bottom)
+
+        // We need additional top margin to account for missing resizing handle when isExpandable is false
```

Comment:
the calculation isn't like resize handle height (depending on it is expandable not expandable) + delta gap?

Reply:
hmm you're right, I should probably expose the current resizing handle height through the sheet controller rather than guessing through these constants.

Reply:
Changed it to what your suggestion is.